spec/simplecov_spec.rb

```diff
@@ -230,10 +230,95 @@
     end
   end
 
+  describe ".collate" do
+    let(:resultset1) do
+      {source_fixture("sample.rb") => {lines: [nil, 1, 1, 1, nil, nil, 1, 1, nil, nil]}}
+    end
+
+    let(:resultset2) do
+      {source_fixture("sample.rb") => {lines: [1, nil, 1, 1, nil, nil, 1, 1, nil, nil]}}
+    end
+
+    let(:resultset_path) { SimpleCov::ResultMerger.resultset_path }
+
+    let(:resultset_folder) { File.dirname(resultset_path) }
+
+    before do
+      expect(SimpleCov).to receive(:run_exit_tasks!)
+    end
+
+    context "when no files to be merged" do
+      it "shows an error message" do
+        captured_output = capture_stderr do
+          glob = Dir.glob("#{resultset_folder}/*.final", File::FNM_DOTMATCH)
+          SimpleCov.collate glob
+        end
+
+        expect(captured_output).to eq("There's no reports to be merged\n")
+      end
+    end
+
+    context "when a single report to be merged" do
+      before do
+        @result1 = SimpleCov::Result.new(resultset1)
+        @result1.command_name = "result1"
+        SimpleCov::ResultMerger.store_result(@result1)
+        FileUtils.mv resultset_path, "#{resultset_path}.final"
+      end
+
+      after do
+        SimpleCov.clear_result
+        FileUtils.rm resultset_path
+        FileUtils.rm "#{resultset_path}.lock"
+        FileUtils.rm "#{resultset_path}.final"
+      end
+
+      it "creates a merged report identical to the original" do
+        glob = Dir.glob("#{resultset_folder}/*.final", File::FNM_DOTMATCH)
+        SimpleCov.collate glob
+
+        expected = {"result1": {coverage: {source_fixture("sample.rb") => {lines: [nil, 1, 1, 1, nil, nil, 1, 1, nil, nil]}}}}
+        collated = JSON.parse(File.read(resultset_path), symbolize_names: true).transform_values { |v| v.reject { |k| k == :timestamp } }
+        expect(collated).to eq(expected)
+      end
+    end
+
+    context "when multiple reports to be merged" do
+      before do
+        @result1 = SimpleCov::Result.new(resultset1)
+        @result1.command_name = "result1"
+        SimpleCov::ResultMerger.store_result(@result1)
+        FileUtils.mv resultset_path, "#{resultset_path}.final"
+
+        @result2 = SimpleCov::Result.new(resultset2)
+        @result2.command_name = "result2"
+        SimpleCov::ResultMerger.store_result(@result2)
+        FileUtils.mv resultset_path, "#{resultset_path}2.final"
+      end
+
+      after do
+        SimpleCov.clear_result
+        FileUtils.rm resultset_path
+        FileUtils.rm "#{resultset_path}.lock"
+        FileUtils.rm "#{resultset_path}.final"
```

Comment:
Same-ish thing for the the after block smells like it might be better as a small helper method here in the spec

Reply:
Addressed by https://github.com/colszowka/simplecov/pull/780/commits/6de9ebf3df3eeb39b4bf16f2403ad98c9d3148e2.